pkg/serverless/invocationlifecycle/trace.go

```diff
@@ -217,3 +230,58 @@ func InjectSpanID(executionContext *ExecutionStartInfo, headers http.Header) {
 		executionContext.SpanID = value
 	}
 }
+
+func capturePayloadAsTags(value interface{}, targetSpan *pb.Span, key string, depth int, maxDepth int) {
+	if key == "" {
+		return
+	}
+	if value == nil {
+		targetSpan.Meta[key] = ""
+		return
+	}
+	if depth >= maxDepth {
+		switch value := value.(type) {
+		case map[string]interface{}:
+			targetSpan.Meta[key] = convertJSONToString(value)
```

Comment:
ahhh cool so as soon as we reach the max depth the remainder value gets converted to a string so we have a json parsed all the way until max depth then we stop recursion.
did i understand this right?

Reply:
Yep.